include/cxx.h

```diff
@@ -297,6 +297,93 @@ class Box final {
 };
 #endif // CXXBRIDGE1_RUST_BOX
 
+#ifndef CXXBRIDGE1_RUST_OPTION
+template <typename T>
+class Option final {};
+
+template <typename T>
+class Option<T&> {
+public:
+  Option() noexcept;
+  Option(Option&&) noexcept;
+  Option(T&) noexcept;
+  ~Option() noexcept;
+
+  Option<T&>& operator=(Option&&) noexcept;
+  const T& operator*() const noexcept;
+  const T* operator->() const noexcept;
+  T& operator*() noexcept;
+  T* operator->() noexcept;
+
+  bool has_value() const noexcept;
+  const T& value() const noexcept;
+  T& value() noexcept;
+  void reset();
+  void set(T&) noexcept;
+  static Option<T&> from_raw(T*) noexcept;
+  T* into_raw() noexcept;
+private:
+  void* repr;
+
+  void drop() noexcept;
+};
+
+template <typename T>
+class Option<const T&> {
+public:
+  Option() noexcept;
+  Option(const Option&) noexcept;
+  Option(Option&&) noexcept;
+  Option(const T&) noexcept;
+  ~Option() noexcept;
+
+  Option<const T&>& operator=(Option&&) noexcept;
+  const T& operator*() const noexcept;
+  const T* operator->() const noexcept;
+
+  bool has_value() const noexcept;
+  const T& value() const noexcept;
+  void reset();
+  void set(const T&) noexcept;
+  static Option<const T&> from_raw(const T*) noexcept;
+  const T* into_raw() const noexcept;
+private:
+  void* repr;
+
+  void drop() noexcept;
+};
+
+template <typename T>
+class Option<Box<T>> {
+public:
+  Option() noexcept;
+  Option(Option&&) noexcept;
+  Option(Box<T>&&) noexcept;
+  ~Option() noexcept;
+
+  Option<Box<T>>& operator=(Option&&) noexcept;
+  const Box<T>& operator*() const noexcept;
+  const Box<T>* operator->() const noexcept;
+  Box<T>& operator*() noexcept;
+  Box<T>* operator->() noexcept;
```

Comment:
If I change this to return `Box<T>&`, then it should automatically dereference down to the T. I'm not sure if that is the expected behavior from a user though